res/enkf/plot_data/ensemble_plot_gen_data.py

```diff
@@ -85,6 +90,13 @@ def getMinValues(self) -> DoubleVector:
         """@rtype: DoubleVector"""
         return self._min_values().setParent(self)
 
+    def copy_realizations(self, data, realizations):
```

Comment:
I presume `data` here is a numpy array? Is it necessary to leave allocation of that up to the consumer of the function, or would it be possible to allocate it inside the Python function body and hence implement `to_numpy_array` instead of `copy_realizations`?

Reply:
Yes data is the numpy array

Did consider doing allocation at part of call, the issue is that currently at the Python level the EnsemblePlotGenData class does not have the knowledge of the "data size" thus can not do the allocation. That information is currenlty exposed via ModelConfig:

https://github.com/equinor/ert/blob/main/res/enkf/export/gen_data_collector.py#L31
https://github.com/equinor/ert/blob/main/res/enkf/export/gen_data_collector.py#L35

It is possible to make EnsemblePlotGenData aware of data size by adding and exposing some extra C functions  (as the underlying enkf_plot_gendata code does have a reference to gen_data_config), if we want that behavior

Reply:
I think I agree, will take a look at the details

Reply:
Have refactored so numpy array is allocated within libres which makes the interface much cleaner and self contained.

As a side of this have converted over to use PyBind instead of cwrap